packages/analytics/src/Consent.ts

```diff
@@ -0,0 +1,79 @@
+import { CONSENT_KEYS } from './utils';
+import merge from 'lodash/merge';
+import pick from 'lodash/pick';
+import type { ConsentData } from './types/analytics.types';
+import type StorageWrapper from './utils/StorageWrapper';
+
+const defaultConsentData: ConsentData = {
+  marketing: false,
+  preferences: false,
+  statistics: false,
+};
+
+/**
+ * Stores the consent values in a new instance.
+ */
+class Consent {
+  storage: StorageWrapper;
+  /**
+   * Constructs a new consent instance with the passed in storage wrapper instance.
+   *
+   * @param storage - The storage wrapper instance where data will be stored.
+   */
+  constructor(storage: StorageWrapper) {
+    // NOTE: For now, we will only check if the storage reference is set to something,
+    //      as we are already validating the storage on the analytics class, so it is not
+    //      required to do it right now. If we expose this class to be used in other
+    //      scenarios, then we will need to revisit this validation code.
+    if (!storage) {
+      throw new Error(
+        'Invalid storage instance provided to Consent constructor',
+      );
+    }
+
+    this.storage = storage;
+  }
+
+  /**
+   * Returns the stored consent object. If not defined, returns the defaults.
+   *
+   * @returns Promise that will resolve with the consent data.
+   */
+  async get(): Promise<ConsentData> {
+    const getItem = (await this.storage.getItem(
+      'consent',
+    )) as Partial<ConsentData>;
+
+    return {
+      marketing: getItem?.marketing || defaultConsentData.marketing,
+      preferences: getItem?.preferences || defaultConsentData.preferences,
+      statistics: getItem?.statistics || defaultConsentData.statistics,
+    };
+  }
+
+  /**
+   * Merges consent default properties with incoming ones from data received.
+   *
+   * @param data - Object with properties for the different types of preferences (Marketing, preferences or statistics).
+   * @returns Promise that will resolve with the instance that was used when calling this method to allow chaining.
+   */
+  async set(data: Partial<ConsentData> = defaultConsentData): Promise<Consent> {
+    const consent = pick(data, CONSENT_KEYS);
+    const currentConsentFromStorage = ((await this.storage.getItem(
+      'consent',
+    )) || {}) as typeof data;
+
+    const newConsent = merge(currentConsentFromStorage, {
+      marketing: consent.marketing || !!currentConsentFromStorage?.marketing,
```

Comment:
Couldn't we directly merge the previous consent with the new one without having these extra validations, like we had before?

Reply:
Yes we could, changed ConsentData to be a Partial by default, so we can do it. Thanks!